iceoryx_utils/include/iceoryx_utils/internal/cxx/expected.inl

```diff
@@ -127,96 +153,119 @@ inline bool expected<ValueType, ErrorType>::has_error() const noexcept
 template <typename ValueType, typename ErrorType>
 inline const ErrorType&& expected<ValueType, ErrorType>::get_error() const&& noexcept
 {
-    return std::move(*m_store.template get_at_index<1>());
+    if (!has_undefined_state())
+    {
+        return std::move(*m_store.template get_at_index<1>());
+    }
+    else
+    {
+        std::cerr << "get_error() called on invalid object" << std::endl;
```

Comment:
I would state it more precisely: `get_error() called on an expected in an undefined state`.